pyccel/ast/builtins.py

```diff
@@ -260,8 +260,6 @@ class PythonFloat(PyccelAstNode):
     _attribute_nodes = ('_arg',)
 
     def __new__(cls, arg):
-        if isinstance(arg.dtype, NativeBool):
-            return PythonInt(arg)
```

Comment:
i was thinking about changing the dtype of the class here

Reply:
So that a `PythonInt` returns a float? That sounds like a documentation nightmare to me

Reply:
I mean changing the dtype instead of casting, but i think we should not modify the object, so it is fase to use it in the printer.

Reply:
Ah, so:
```
def __new__(cls, arg):
    arg.dtype=float
    return arg
```
2 counter examples:
Example 1:
```
a = 2
b = float(a)
```
You don't want to change the dtype of the `Variable` a

Example 2:
```
c = float(a+b)
```
if `a+b` is a complex and you just change the type of `a+b` then you will print:
```
float c = a + b
```
and you will have warnings about truncation

Reply:
it will look weird but here is what i meant:
```python
        if isinstance(arg.dtype, NativeBool):
            cls._dtype = NativeFloat()
```